web/src/stream_events.js

```diff
@@ -146,6 +149,19 @@ export function mark_subscribed(sub, subscribers, color) {
     // update navbar if necessary
     message_view_header.maybe_rerender_title_area_for_stream(sub);
 
+    if (stream_create.get_name() === sub.name) {
+        // In this case, we just created this channel using this very
+        // browser window. We redirect the user to that channel so
+        // they can use the channel that they just created.
+        //
+        // It's important that we do this here, not in
+        // add_sub_to_table, to avoid showing or flashing a subscriber
+        // bookend during the window that the client doesn't yet know
+        // that we're a subscriber to the new channel.
+        stream_create.reset_created_stream();
+        browser_history.go_to_location(hash_util.by_stream_url(sub.stream_id));
+    }
```

Comment:
I rewrote this comment. A few notes on what I'm changing:
- Trying to consistently use `channel` in comments. (I know this will get easier once we start bulk-migrating variable names).
- Placing the comment explaining the purpose of an `if` block inside the block. I think this is usually more readable.
- Providing a bit more context on the motivation for the behavior; this saves time when trying to understand the code later.

``` diff
diff --git a/web/src/stream_events.js b/web/src/stream_events.js
index d3bfc380ac..cab1b3ecb3 100644
--- a/web/src/stream_events.js
+++ b/web/src/stream_events.js
@@ -149,15 +149,16 @@ export function mark_subscribed(sub, subscribers, color) {
     // update navbar if necessary
     message_view_header.maybe_rerender_title_area_for_stream(sub);
 
-    // Note: It is important to perform the 'redirect to channel view'
-    // operation here i.e. after subscribing to the channel, because
-    // doing so in 'add_sub_to_table' results in fetching messages in narrow
-    // before subscription hence a buggy "You subscribed to..." bookend.
     if (stream_create.get_name() === sub.name) {
-        // The `stream_create.get_name()` check tells us whether the
-        // stream was just created in this browser window.
+        // In this case, we just created this channel using this very
+        // browser window. We redirect the user to that channel so
+        // they can use the channel that they just created.
+        //
+        // It's important that we do this here, not in
+        // add_sub_to_table, to avoid showing or flashing a subscriber
+        // bookend during the window that the client doesn't yet know
+        // that we're a subscriber to the new channel.
         stream_create.reset_created_stream();
-        // Go to the newly created stream interleaved view.
         browser_history.go_to_location(hash_util.by_stream_url(sub.stream_id));
     }
 
```